tensorflow/python/ops/image_ops.py

```diff
@@ -242,6 +242,26 @@ def _CheckAtLeast3DImage(image):
                      image.get_shape())
 
 
+def _slice_channels(image):
+  """Slices a tensor into one tensor for each 'channel' (the last dimension).
+
+  Args:
+    images: Tensor containing the images to be sliced.
+
+  Returns:
+    list of tensor, one for each channel.
+  """
+
+  image_shape = image.get_shape()
```

Comment:
Can you use tf.rank here? That would work also on tensors whose rank not statically defined.